lib/bus.js

```
@@ -342,7 +342,8 @@ class MessageBus extends EventEmitter {
           }
         };
       }
-      this._connection.message(msg);
+      if(this._connection.stream.writable)
+        this._connection.message(msg);
```

If it's not writable, you seem to be ignoring the message. You have to handle that case.

---

Which message are you referring to?

---

The message that should be sent. It should probably throw an error.

---

I'm not sending a message, you are. In _removeMatch()

---

I'm just removing an event listener, which apparently results in asynchronous behavior with no catch()